src/components/donation/Steps/common/ProgramSelector.tsx

```diff
@@ -46,17 +51,27 @@ export function ProgramSelector({
           </>
         )}
       </Listbox.Button>
-      <Options endowId={endowId} classes="absolute top-full left-0 z-10" />
+      <Options
+        endowId={endowId}
+        classes="absolute top-full left-0 z-10"
+        onOptionsLoaded={(options) => {
+          const selectedProgram = options.find((o) => o.id === program.value);
+          if (!selectedProgram) return;
+          onChange({ label: selectedProgram.title, value: selectedProgram.id });
```

Comment:
[initial assumption](https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/3037/files#diff-ebbb280127392b403c6bca1a44a99b8206da71ecaa34482aaed5357b278b8085R72) regarding `by="value"` is incorrect as options are async (initially empty) so initial program value can't be matched